src/components/cards/Comments.js

```diff
@@ -1,14 +1,76 @@
-import React from "react";
-// import AccountCircleIcon from "@material-ui/icons/AccountCircle";
+import React, { useState } from "react";
 import styled from "styled-components/macro";
+import Dialog from "@material-ui/core/Dialog";
+import DialogContent from "@material-ui/core/DialogContent";
+import DialogTitle from "@material-ui/core/DialogTitle";
+
+export default function Comments({
+  id,
+  showComments,
+  comments,
+  handleDelete,
+  handleEdit
+}) {
+  const [editFormDialog, setEditFormDialog] = useState(false);
+  const [editComment, setEditComment] = useState("");
+  const [cardID, setCardID] = useState("");
+  const [commentID, setCommentID] = useState("");
 
-export default function Comments({ id, showComments, comments, handleDelete }) {
-  // console.log(showComments);
   function handleDeleteClick(id, deleteComment) {
-    // console.log(id, deleteComment);
     handleDelete(id, deleteComment);
   }
 
+  function handleEditClick(id, commentData) {
+    setCommentID(commentData._id);
+    setEditComment(commentData.comment);
+    setCardID(id);
+    setEditFormDialog(true);
+  }
+
+  function handleEditCommentSubmit(event) {
+    event.preventDefault();
+    const form = event.target;
+    const formData = new FormData(form);
+    const data = Object.fromEntries(formData);
+    data._id = commentID;
+    handleEdit(cardID, data);
+  }
+
+  function handleClose() {
+    setEditFormDialog(false);
+  }
+
+  function editForm() {
```

Comment:
You could put this component in a separate file. You generally have very long files.